### ISSUE TITLE
Boom sound when using audiodxe

### ISSUE THREAD

**Issue by franksfc** [acidanthera/bugtracker#2073]
I have a dell Inpisron 3670 machine with alc867/891. When I use the audiodxe and play the sound, there is always a boom sound before Duang. The speaker is not to be blamed.

[opencore-2022-06-26-043816.txt](https://github.com/acidanthera/bugtracker/files/9050606/opencore-2022-06-26-043816.txt)

**Comment by franksfc** [comment]
Thank you for your help! But I tried the arguments with no luck, including --gpio-setup, --restore-nosnoop and --gpio-pins 0~15. Maybe I haven't understood the configuration.pdf correctly. Could you give me more hints?

**Comment by mikebeaton** [comment]
It's helpful to know the older version has the same problem, thanks. I'll have to leave this open for now and get back to you in due course, unless any other devs have time to look sooner. (Btw 'Duang', you do mean 'chime', yes?)

**Comment by franksfc** [comment]
Yes, Duang means the chime.

**Comment by mikebeaton** [comment]
@franksfc Have you tried `SetupDelay`? Depending on your exact meaning of 'boom', possibly you just need this setting. Some sound card amplifiers do not respond immediately to volume changes. Therefore on card init the card can be at max (or other very loud 'default') audio volume initially, and for up to 0.5 after we change the amplifier settings. This introduces a delay before playing the chime to allow for this.

**Comment by franksfc** [comment]
Unfortunately, no. I noticed 'in microseconds' so I tried 1000,10000,500000,1000000,10000000 and the issue didn't improve at all. I have also tried resettrafficclass and discconecthda quirks. Forgive me if the sentences are grammatically wrong as I am a beginner in English.

**Comment by mikebeaton** [comment]
Yes, it is microseconds, and the docs suggest up to 0.5s delay, so your values make sense. I assume with the 10,000,000 value you did actually observe a long (10s) delay before the sound plays? If so, is the 'boom' still immediately before the sound, or ~10s before?

**Comment by franksfc** [comment]
it booms 10s before the chime

**Comment by franksfc** [comment]
Thanks! I will always be there if you need any further information.

**Comment by MagistraNocte** [comment]
Hi, just wanted to say I'm seeing a similar problem too, I'm working on a Powermac G4 mod and I wanted to have an internal speaker to play the boot chime like the original Powermac G4, so i set up a small speaker attached to a small PAM8403 amplifier powered through USB (i actually give it power through the SPDIF motherboard header with some jumper cables, but I also tried giving it the 5V through the USB 1/2 header, the led header, molex and even an external source through my arduino board, the problem is still there, it's not in the power source) connected through a case front panel audio cable to the fronto panel audio header on my motherboard (Gigabyte Z690M DS3H), as soon as the boot chime starts playing I hear a loud pop (the chime itself starts with a loud pop, as if the sound card were abruptly being turned on), then, later in the boot process close to the login screen showing up I heard a couple of other loud pop as if the speakers were powered off and on again or something like that, I ruled out the small speaker or the PAM8403 being the problem since connecting my main speakers to the same motherboard audio header gives the same problem (the pops are not as louds with my speakers but that's probably related to the amplifier i'm driving them with smoothing it up a bit, but the pattern of the pops is exactly the same). it's also notable to add that the problems does not occur when I connect any audio source to my motherboard's rear panel audio out instead of the front panel header, it also doesn't occur when disabling the boot chime in the config.plist (or setting the gain low enough for it to not play, for that matters) so it seems to be related to opencore and the boot chime, but also specifically to the Intel HDA motherboard header channel.

**Comment by MagistraNocte** [comment]
Ok, I tried adding a delay with SetupDelay and I can confirm too that the first loud pop happens as soon as the motherboard logo disappear regardless of the delay set (with a delay set the chime is pop-less but the two pops near the login screen are still there), also weirdly enough SetupDelay in my version of opencore (0.8.5) doesn't seem to be in microseconds, a value of 10 is an audible delay, a value of 1000 are more or less 10 seconds and a value of 1000000 (which would be 1 second if it were in microseconds) doesn't let me boot at all (I guess it would have booted but it's reading it as a 10000 seconds delay), should I open a separate issue for the SetupDelay bug?

**Comment by MagistraNocte** [comment]
> I don't believe this is affecting many other systems

Could this be because it is related to the front panel motherboard header only, and that is not somethin every users is using? especially not to play the boot chime since front panels usually only have headphones ports and it's not usual for users to already have headphones on before even booting. maybe the issue is actually affecting more systems than previously tought, if not all (in my case it's just yout average alder lake desktop build, nothing peculiar with weird layouts like a laptop).
Can anyone confirm there is no pop/boom/crackle sound during boot with the boot chime enabled (and outputted to all the ports) coming out through your front panel headphone port in your system?

**Comment by MagistraNocte** [comment]
Also worth noting the problem is also present when booting to windows, but since opencore is still used even when booting to windows and the boot chime still plays, this is still probably opencore related

**Comment by mikebeaton** [comment]
> Also worth noting the problem is also present when booting to windows, but since opencore is still used even when booting to windows and the boot chime still plays, this is still probably opencore related

Well, that should be easy to check: on almost all systems (and I am sure on a standard Dell - I assume you are reporting about the same hardware, since you do not say?) you can boot into Windows directly, not via OpenCore (if you set up a native BIOS boot entry for it, if there is not one already).

You can also try using OC, but with and without AudioDxe enabled (obviously when disabled losing your OC boot chime, if you have that set up), to see if the presence of that is what is making the difference.

**Comment by MagistraNocte** [comment]
Hi, thanks for the reply, I already said in my first comment that disabling the boot chime (or setting its gain so low that it doesn't play) the issue doesn't occur.

**Comment by MagistraNocte** [comment]
> and I am sure on a standard Dell - I assume you are reporting about the same hardware, since you do not say?

I think you missed my other comments.

**Comment by mikebeaton** [comment]
If the boom happens after boot chime, but not after no boot chime, then sure, the setup which AudioDxe is doing is indeed causing the issue. (Even more so, if we find no boom when AudioDxe is completely disabled in the Drivers section - could you try this? AudioDxe does a bunch of setup when installed, even if it never plays a sound afterwards.) It is very hard to fix these issues with no machine to replicate on though, it really requires someone with some knowledge of audio programming and OC programming, and a machine to try changes on.

**Comment by MagistraNocte** [comment]
I can confirm no boom is heard with AudioDxe disabled, even though the little speaker itself is still powered by the motherboard (i can hear the background noise when up close).

**Comment by MagistraNocte** [comment]
my machine is up for testing anything if you need it, audio is my field of work but I'm no programmer, I doubt I could help in that regard unfortunetely.

**Comment by MagistraNocte** [comment]
with actual headphones connected to a case front panel connected to the f_panel audio header on the motherboard I hear the boot chime but the booms are not really booms anymore, they're way quieter than with the amplified speaker, I can identify them just because I know they're there, but they're barely audible, it's clear that opencore is not the only part of the equation, maybe sometginh to do with impedence difference between headphones and a 4ohm/3watts little speaker? yet the same speaker/amplifier combo connected to the rear panel audio out doesn't play any boom sound

**Comment by MagistraNocte** [comment]
> this is only a spare-time pursuit, unfortunately - so I can't promise anything especially timely, I'm afraid.

No worries, I'm just glad you're taking time to address this issue, thank you very much.
I produced the log you requested (if you need a differen "Target" value in config.plist just let me know, set that to 67 for now), and I also recorded 3 videos of the issue (sorry for the mess! I'm still working on the case mod): the first one with the boot chime and the booms playing (I have highlighted the exact moment of every boom with onscreen text, I hope it's clear enough), the second one with same setup but with the screen in the picture so that hopefully you can match the exact moment the boom takes place with a specific line in the logs (I've noticed the second and third booms corresponds to lines that are not in the log file, right before all those "private" lines, I guess it is because at that time opencore has done its thing and the os has taken on, and so that part of the verbose is logged to some other file inside the OS? if this is the case just let me know where to find it and I will provide it), the third one with no boot chime and no boom sound (here "MaximumGain" is set to -100, previously it was -30). I also set "SetupDelay" to 2000 so that the first boom doesn't get lost in the chime sound (since it seems to be fainter then the two later ones). I also included my config.plist.
This is my current setup:

- Opencore version: 0.8.5 (sorry I haven't taken the time to update to 0.8.6 yet, if this is critical just let me know)
- Motherboard: Gigabyte Z690M DS3H (audio codec: Realtek ALC897)
- CPU: i7 12700k alder-lake
- GPU: Sapphire AMD 5700XT (the founder edition looking one, not the pulse or the nitro one)
- small 4ohm/3watts speaker driven by a PAM8403 amplifier chip connected to the motherboard F_audio header (same happens using a real case front panel module with actual speakers connected to it through normal audio jack cable, but it does not happen if connected to the rear motherboard audio out, see my previous comments).

I don't think the rest of the setup is relevant, if it is just let me know.
Thank you again for your time.
[Files](https://drive.google.com/file/d/1RubsvZVWR8r1XPS2m__Bqjl6tDwx-uEv/view?usp=share_link)

**Comment by mikebeaton** [comment]
Hi @MagistraNocte - I have put up a test build at https://github.com/acidanthera/OpenCorePkg/actions/runs/3613166325 - please can you download and run the debug version of this.

I have added some additional two second delays during audio setup to try to determine exactly which audio setup command(s) are causing the unwanted noise(s). (It should take about 20 extra seconds to start up - it's adding delays on 2 or 3 different commands, on 2 or 3 different widgets - so be patient.)

I need to know, on the version with the chime which gives clicks or booms, exactly when before the chime the first click or boom sound (or sounds, if there are more than one) occur(s). (E.g. on the video you sent, the initial click/boom was 2 seconds before the start of the chime.) You already have Audio->SetupDelay=2000 in your config, as you mentioned - that was helpful for being sure where the issue occurs; so that I know what delays I am seeing where, just leave that as it is.

Another annotated video with the screen in shot would be very helpful (the first boom - which I am sure you can hear clearly there - is not very audible in the video, so the annotations are useful), plus the associated OC debug log.

**Comment by MagistraNocte** [comment]
Hi @mikebeaton sorry for the late response, I wasn't home.
Thank you again for putting time into this.
[Here](https://drive.google.com/file/d/1cP06JYSk4_Y9yyN8E_Jf0D9ulNsoyMbb/view?usp=share_link) is another video (screen in shot) with text highlights (I managed to put the speaker right next to the screen this time, you should hear everything clear now) and the corresponding opencore log.
If you need anything else just let me know.

**Comment by mikebeaton** [comment]
Okay, that's great. Just to confirm that you are seeing/hearing what I am: there's one click immediately on setting vref 1 on widget 0x1B (and nothing else before or around the chime), and then two clicks just before the macOS login screen is displayed.

The trick will be the setup order, there'll be something else which will ramp up the power more slowly, avoiding the click, and which should be done after enabling the widget vref instead of before, as now. (And the reverse would need doing before the OS starts.) When I get a few more minutes to look at this, I'll hunt for plausible candidates.

**Comment by MagistraNocte** [comment]
> Okay, that's great. Just to confirm that you are seeing/hearing what I am: there's one click immediately on setting vref 1 on widget 0x1B (and nothing else before or around the chime), and then two clicks just before the macOS login screen is displayed.

The first few lines happens fast, it's difficult to tell, looks like it can be anything between "Starting to play chime..." and "Setting vref 1" but it makes sense it is this last one. I can confirm the one you listed are the only clicks I hear.

> The trick will be the setup order, there'll be something else which will ramp up the power more slowly, avoiding the click, and which should be done after enabling the widget vref instead of before, as now. (And the reverse would need doing before the OS starts.) When I get a few more minutes to look at this, I'll hunt for plausible candidates.

Again thank you. I understand this issue is not the priority at all, and I really appreciate your work on this.

**Comment by mikebeaton** [comment]
@MagistraNocte Sorry it's taken me so long to get back to this. I don't think you ever put up a SysReport, please can you set SysReport=true in config.plist, then reboot, and send me the contents of {ESP}/SysReport/Audio folder.

**Comment by MagistraNocte** [comment]
> Sorry it's taken me so long to get back to this

No worries.
Here's the requested folder, I used the same debug efi folder I used for past logs and test regarding this issue (I kept a copy), everything should be the same as in previous messages, let me know if you need anything else.
[Audio.zip](https://github.com/acidanthera/bugtracker/files/10975009/Audio.zip)

**Comment by mikebeaton** [comment]
@MagistraNocte - Thanks for that dump. I was playing around with this a bit more last night, and I'm a bit less hopeful.

I can recreate exactly the same booms and pops if using the earphone output from the built in Realtek ALC255 on my Dell 3070 i9 MFF.

I previously understood @franksfc's original report to state that there were no booms and pops at all without AudioDxe (i.e. with that driver disabled), but that is not my experience:

|   | Without AudioDxe | With AudioDxe (+ SetupDelay*) |
| --- | --- | --- |
| Pops on starting machine | yes | yes |
| Additional pop before chime | no (n/a) | yes |
| Two pops during macOS boot progress bar | yes | yes |
| Pops while shutting down macOS | yes | yes |
| Pops while starting Linux | yes | yes |

*Using `SetupDelay` just helps to separate the audio setup pop from the chime itself.

@MagistraNocte - Can you confirm (or deny) any of the above findings, for your setup?

I also attempted fairly radically changing the setup order - and unfortunately got exactly the same pop, on exactly the same command (i.e. when enabling output amp vref as per the OC log with delays). You can try this alternative version too, if you like, it will be the macOS build artefact from here https://github.com/mikebeaton/OpenCorePkg/actions/runs/4423925293, when it has finished building.

**Comment by MagistraNocte** [comment]
> @MagistraNocte - Thanks for that dump. I was playing around with this a bit more last night, and I'm a bit less hopeful.
> 
> I can recreate exactly (?) the same booms and pops if using the earphone output from the built in Realtek ALC255 on my Dell 3070 i9 MFF.
> 
> I previously understood @franksfc's original report to state that there were no booms and pops at all without AudioDxe (i.e. with that driver disabled), but that is not my experience:
> 	Without AudioDxe 	With AudioDxe (+ SetupDelay*)
> Pops on starting machine 	yes 	yes
> Additional pop before chime 	no (n/a) 	yes
> Two pops during macOS boot progress bar 	yes 	yes
> Pops while shutting down macOS 	yes 	yes
> Pops while starting Linux 	yes 	yes
> 
> *Using `SetupDelay` just helps to separate the audio setup pop from the chime itself.
> 
> @MagistraNocte - Can you confirm (or deny) any of the above findings, for your setup?
> 
> I also attempted fairly radically changing the setup order - and unfortunately got exactly the same pop, on exactly the same command (i.e. when enabling output amp vref as per the OC log with delays). You can try this alternative version too, if you like, it will be the macOS build artefact from here https://github.com/mikebeaton/OpenCorePkg/actions/runs/4423925293, when it has finished building, so another video with audio, showing the log on screen and being able to hear when it pops, would be useful!

I can confirm that even with this new test version I get the same results as before:
| | Without AudioDxe | With AudioDxe (+ SetupDelay*) |
| - | - | - |
| Pops on starting machine | yes (this is probably normal and anavoidable because it happens when the audio components first receive power? that's why I never bringed it up) | yes |
| Additional pop before chime | no (no chime) | yes |
| Two pops during macOS boot progress bar | no | yes |
| Pops while shutting down macOS | no (there is one if I'm restarting it but that is the "Pops on starting machine" one) | no |
| Pops while starting Windows | yes (when starting the machine); no (the one before the chime, because no chime) | yes |

Made another video of the boot with the same settings as the last one + SysReport=true and this time I used the time lapse function of my iphone to better see the lines at wich the 2nd and 3rd booms occurs, it seems the 2nd one appears somewhere around the `GTrace synchronization point 1` and the `Unsupported PCH` lines, while the 3rd appears right in the middle of all those `<private>` lines. As for the first one: with this test version it appears when the screen is waiting on the bunch of lines that end up with `OCC: Install console control (2C17A0B0/0/0), Current - Success`.
[Here](https://drive.google.com/file/d/1XS1LLiPn6P1Um92yhTux1GcNmsONOFWn/view?usp=share_link) is a complete copy of my test EFI folder (with sanitized config.plist), the video, the opencore log for the boot seen in the video, and the SysReport audio folder
Thank you.

**Comment by mikebeaton** [comment]
Could you see if you can recreate the situation in your first videos, where debug output carries on after the line `OCC: Install console control (.....), current - Success`? (As it is those lines which help to show exactly when the error is occurring - make sure you install a DEBUG version of everything - well, at least, of AudioDxe as well as OpenCore.)

**Comment by MagistraNocte** [comment]
> Could you see if you can recreate the situation in your first videos, where console debug output carries on after the line `OCC: Install console control (.....), current - Success`? As it is seeing the `HDA: ` log lines on screen compared to the timing of the sounds which helps to show exactly when the issue is occurring. Make sure you install a DEBUG version of everything from that alt. build - well, at least, of AudioDxe.efi as well as OpenCore.efi. (Also, maybe try `BuiltinText` instead of `BuiltinGraphics`, etc. - I can't immediately see why this has changed, as you do seem to be using the debug version of AudioDxe.efi: the `HDA: ` lines are in the text log, assuming the text log comes from the same run as the video.)

Yes, the txt log comes from the same run as the video, I'm not sure why the logging changed, I'm still using the same EFI folder I used for the old videos (I kept an untouched copy of it while using another non-debug one for everyday use), and yes I'm using the debug version of everything (or at least everything I'm using in this EFI folder) that was in the test build you sent me, I was under the assumption that it was logging differently because of fome change on your part.

**Comment by MagistraNocte** [comment]
Ok changing to BuiltInText reverted it back to how it was in previous videos, strange though, I don't remember ever changing it in the firts place, anyway, [here](https://drive.google.com/file/d/1LaxA5WU6Cs90FoNnqS99zk8pip4ib8Ta/view?usp=share_link)'s another video, the chime boom clearly occurs at the `enable output amp vref 1 - Success` line, the rest is same as my previous comment.

**Comment by MagistraNocte** [comment]
One thing to note: you tested using headphones connected directly to the motherboard, in my current setup instead I have a small speaker (+ amplifier) connected to the hd_audio header on the motherboard (the one you usually connect case front panel to) BUT it is connected through just some jumper cables, i don't have an actual front panel cable connected to it, I'm just using the audio + and - pins to extract the signal to send to the speaker, the rest of the hd_audio header is free, as a result it is not seen as a possible output device by the OS (not even windows) because I have not populated the other pins through which the OS knows that a front panel exists (if you have one) and that headphones are connected to it, so I cannot play any sound through the speaker after boot, AudioDxe on the other hand seems to not care about the sense pin and all and it is able to play the chime through it just fine (although with clicks).

**Comment by mikebeaton** [comment]
Yes, AudioDxe does not check the sense pins. Making it do so would be a possible upgrade.

Given the ubiquity of pops, clicks and booms (with sufficient amplification) on these sound cards, I think this would have to be moved to low priority. Sorry!

A more like-for-like test would be if you can configure macOS so that it starts up and plays sounds through the very same channel used for the chime on your rig. I'm guessing (as I think you may be suggesting?) that it might end up sounding like my system (i.e. clicks during macOS startup even without AudioDxe).

**Comment by mikebeaton** [comment]
> I was suggesting you should see if you hear the clicks through headphones connected to your case front pane audio out

My headphones _are_ connected to my case front panel audio out! (You just mean the normal headphone connector on the front of the case - in a situation like mine with a standard setup - I suppose.)

**Comment by MagistraNocte** [comment]
> My headphones _are_ connected to my case front panel audio out! (I assume you just mean the normal headphone connector on the front of the case - in a situation like mine with a standard setup.)

Oops! i guess I misread your comment, then your test shuold be equivalent to my current setup

**Comment by mikebeaton** [comment]
> Oops! i guess I misread your comment, then your test shuold be equivalent to my current setup

Well I suppose not quite, because when my headphones are plugged in macOS carries on to produce sound through those, whereas in your case it produces sound through another channel, right?

**Comment by MagistraNocte** [comment]
In my case macOS outputs sound through the usb sound card (just because I set that in system preferences, I can switch to motherboard out and optical, which I don't have), but only after boot, during boot AudioDxe is set to output sound to the hd_audio header only

**Comment by mikebeaton** [comment]
Yes that is what I understood. The additional test which I would be interested in is, if you could manage to persuade macOS to output sound through the same channel of the same codec as your chime, then does your system replicate everything on my system, including two clicks at macOS startup (even if you do not use AudioDxe)?

**Comment by MagistraNocte** [comment]
You mean having an actual front panel connected to the motherboard instead of a couple of jumper cables and selecting it as the audio device in the OS prior to rebooting? I think I did it back before my first comment, I don't quite remember but I could certainly do that again

**Comment by MagistraNocte** [comment]
What pin should I short the sense one with?
this is the pin configuration on my board:

![Schermata 2023-03-17 alle 2 35 17 PM](https://user-images.githubusercontent.com/38999587/225920099-5b454866-1bbd-4d26-9915-d768d40fbed7.png)

**Comment by MagistraNocte** [comment]
Actually I recall reading somewhere that shorting the sense pin only tells the OS that a headphone has been connected in the port, not that there is a front panel connected to the motherboard, I think that is more complicated and requires putting specific resistors between some pin

**Comment by mikebeaton** [comment]
Tbh at a complete guess, I would have said you might just need to short Head Phone Detection to GND. But that is only if your macOS things there _is_ a headphone jack, but just with nothing connected to it. Might be worth a try? If not, I agree it may be more complicated.

**Comment by MagistraNocte** [comment]
I think I'll cut through it and just connect my old case front panel like I did before, I'll do that tomorrow and let you know

**Comment by mikebeaton** [comment]
It would help if you could summarize the tests and booms in a table, so maybe TEST1-TEST6 across the top, and a description of the various booms down the side (a bit like my table, but more entries, I guess!) - and do not leave out the to USB/non-small speaker clicks and pops - you can just describe which speaker they come from in the description.

**Comment by MagistraNocte** [comment]
I had originally started noting all the main speakers booms in the videos and in the comment too, but stopped and erased them after I found out they also occur when unplugging and plugging back the USB soundcard while the system is on (this makes me think it is not related to the original issue, even though I just tested with windows and this doesn't occur), I won't go back and redo it because it was taking a huge time and their timing was not as accurate as the others so the lines they were appearing at were often different between tests, but I con do a table no problem:

| Boom N°                                                          | TEST 1: Front panel, AudioDxe enabled, Device: headphones | TEST 2: Front panel, AudioDxe disabled, Device: headphones | TEST 3: Front panel, AudioDxe disabled, Device: USB soundcard | TEST 4: Front panel, AudioDxe enabled, Device: USB soundcard | TEST 5: No front panel, AudioDxe enabled, Device: USB soundcard | TEST 6: No front panel, AudioDxe disabled, Device: USB soundcard |
|------------------------------------------------------------------|--------------------------------------------------------|---------------------------------------------------------|------------------------------------------------------------|-----------------------------------------------------------|--------------------------------------------------------------|---------------------------------------------------------------|
| 1st (small speaker) the one before the chime                     | Yes (fainter)                                          | No                                                      | No                                                         | Yes (fainter)                                             | Yes                                                          | No                                                            |
| Chime                                                            | Yes                                                    | No                                                      | No                                                         | Yes                                                       | Yes                                                          | No                                                            |
| 2nd (small speaker) the one prior to the `<private>` lines         | Yes                                                    | No                                                      | No                                                         | Yes                                                       | Yes                                                          | No                                                            |
| 3rd (small speaker) the one in the middle of the `<private>` lines | Yes                                                    | No                                                      | No                                                         | Yes                                                       | Yes                                                          | No                                                            |
| 4th (main speakers) comes at the end of the `<private>` lines      | Yes                                                    | Yes                                                     | Yes (double*)                                              | Yes (double*)                                             | Yes                                                          | Yes (comes a little earlier, around where the 3rd usually occurs)                                                          |
| 5th (main speakers) comes a bit after the 4th                    | Yes                                                    | Yes                                                     | Yes                                                        | Yes                                                       | Yes                                                          | Yes (comes a little earlier)                                                          |
| 6th (small speaker) [ex 4th] comes a bit before the black screen | Yes                                                    | No                                                      | No                                                         | Yes                                                       | No                                                           | No                                                            |
| Screen goes black                                                | –                                                      | –                                                       | –                                                          | –                                                         | –                                                            | –                                                             |
| 7th (small speaker) [ex 5th] comes while the screen is black     | Yes                                                    | No                                                      | No                                                         | Yes                                                       | No                                                           | No                                                            |
| 8th (main speakers) comes while the screen is black              | Yes                                                    | Yes                                                     | Yes                                                        | Yes                                                       | Yes                                                          | Yes (come a little earlier, before the screen goes black)                                                          |
| 9th (main speakers) comes while the screen is black              | Yes                                                    | Yes                                                     | Yes                                                        | Yes                                                       | Yes                                                          | Yes (comes a little earlier)                                                           |
| Login screen appears                                             | –                                                      | –                                                       | –                                                          | –                                                         | –                                                            | –                                                             |
| 10th (main speakers) comes during the login screen               | Yes                                                    | Yes                                                     | Yes                                                        | Yes                                                       | Yes                                                          | Yes                                                           |
| 11th (main speakers) comes during the login screen               | Yes                                                    | Yes                                                     | Yes                                                        | Yes                                                       | Yes                                                          | Yes                                                           |

*In this setup it is not a single boom but rather two distinct booms but very close to each other so that they are discernible only through the slo-mo function of the video, in person they sound like a single noise, although qualitatively different than the others.

Here is an overall scheme of the boom pattern regardless of the test considered:

| Speaker              | Small speaker                                              | Main speakers                                                                                                               |
|----------------------|------------------------------------------------------------|-----------------------------------------------------------------------------------------------------------------------------|
| 1st boom             | Only when AudioDxe is enabled                              |                                                                                                                             |
| Chime                | Only when AudioDxe is enabled                              |                                                                                                                             |
| 2nd boom             | Only when AudioDxe is enabled                              |                                                                                                                             |
| 3rd boom             | Only when AudioDxe is enabled                              |                                                                                                                             |
| 4th boom             |                                                            | Always (double when in the front panel setup AND USB soundcard is selected in system preferences) (comes earlier in TEST 6) |
| 5th boom             |                                                            | Always (comes earlier in TEST 6)                                                                                            |
| 6th boom             | Only when AudioDxe is enabled AND in the front panel setup |                                                                                                                             |
| Screen goes black    | –                                                          | –                                                                                                                           |
| 7th boom             | Only when AudioDxe is enabled AND in the front panel setup |                                                                                                                             |
| 8th boom             |                                                            | Always (comes earlier in TEST 6, before the screen goes black)                                                              |
| 9th boom             |                                                            | Always (comes earlier in TEST 6)                                                                                            |
| Login screen appears | –                                                          | –                                                                                                                           |
| 10th boom            |                                                            | Always                                                                                                                      |
| 11th boom            |                                                            | Always                                                                                                                      |

**Comment by franksfc** [comment]
> @MagistraNocte - Thanks for that dump. I was playing around with this a bit more last night, and I'm a bit less hopeful.
> 
> I can recreate exactly (?) the same booms and pops if using the earphone output from the built in Realtek ALC255 on my Dell 3070 i9 MFF.
> 
> I previously understood @franksfc's original report to state that there were no booms and pops at all without AudioDxe (i.e. with that driver disabled), but that is not my experience:
> 
> Without AudioDxe	With AudioDxe (+ SetupDelay*)
> Pops on starting machine	yes	yes
> Additional pop before chime	no (n/a)	yes
> Two pops during macOS boot progress bar	yes	yes
> Pops while shutting down macOS	yes	yes
> Pops while starting Linux	yes	yes
> *Using `SetupDelay` just helps to separate the audio setup pop from the chime itself.
> 
> @MagistraNocte - Can you confirm (or deny) any of the above findings, for your setup?
> 
> I also attempted fairly radically changing the setup order - and unfortunately got exactly the same pop, on exactly the same command (i.e. when enabling output amp vref as per the OC log with delays). You can try this alternative version too, if you like, it will be the macOS build artefact from here https://github.com/mikebeaton/OpenCorePkg/actions/runs/4423925293, when it has finished building, so another video with audio, showing the log on screen and being able to hear when it pops, would be useful!

Here is a recording of my issue. I set my SetupDelay to 1000. There is always a boom sound before the driver tries to play the chime. I know the boom sound may happen in some cases such as booting the PE, but it never happens when booting the windows. Hope I have made it clear.
[Chime 2.zip](https://github.com/acidanthera/bugtracker/files/11471069/Chime.2.zip)

**Comment by mikebeaton** [comment]
Thanks for the recording. It is accepted that there is an issue, which will be annoying to people on whose setup it shows. In fact I was able to replicate something very similar on my headphone jack. But it's likely to be a fairly significant investment of time to figure out how to resolve this (and very likely in some systems it will not be fully resolvable). 'Help wanted' label still applies, for anyone who has, or wants to gain, the necessary skills to look at this.
